### PR TITLE
Jupyter Server Interactive Sessions

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
To make it easier to get started with QCFractal we have had many requests for the ability to provide interactive sessions so that we can build tutorials around this. Originally I believed that we could have built an async client that would use FractalSnowflake and build synchronous calls from there. After talking with Jupyter a bit with @Carreau (Jupyter core) I gave up this tack and simply switched over to a background process.

Instead of going through snowflake we go through a canonical `qcfractal-server` the reason that we do this is that we are able to control the mongo/postgres temporary directory and guarantee that these processes shut down (though this reminds me that this doesn't work on windows sadly). The subprocess approach does loose some interactivity such as `server.await_...()` kinds of calls, but this is something we could potentially write in by using a client to constant ping and check different task queues.

I think in Py3.8+, future changes to Jupyter (thanks @Carreau once more for working on this!), and/or figuring out exactly how `Dask.distributed` does this we might be able to revert. However, the current implementation is relatively robust and we can move on with things for now.

Accomplish items:

- [x] Attempts to write and destroys a async client.
- [x] Adds a FractalSnowflakeHandler for Jupyter (a mouthful, happy to take name suggestions)
- [x] Tweaks the docs slightly attempting to delineate between a full server, a snowflake, and a snowflake handler. I think this will need more work and unfortunately makes things more complex than desired.
- [x] Tweaks docs here and there.
- [x] Write a short quick start tutorial using the new tech and add under getting started docs.

## Discussion
Any thoughts on the following:
- [ ] Consider hiding `FractalServer` in the python API as this should only be called via Python.
- [ ] Better name than FractalSnowflakeHandler for the Jupyter version of this.
- [ ] Should we replace the current Snowflake with `FractalSnowflakeHandler` tech? The current snowflake is very useful for testing however.

## Status
- [ ] Changelog updated
- [ ] Ready to go

### CODE REVIEW

**Comment by dgasmith** [comment]
@Lnaden Ready for review!

File: docs/qcfractal/source/setup_server.rst [hunk 1]
```diff
@@ -58,3 +50,49 @@ up with compute with the following lines:
...
+
+.. warning::
+
+    All data inside a ``FractalSnowflake`` is temporary and will be deleted when the
+    ``FractalSnowflake`` shuts down.
+
+.. code-block:: python
+
+    >>> from qcfractal import FractalSnowflake
+    >>> from qcfractal.int
```
Comment (Lnaden) [inline docs/qcfractal/source/setup_server.rst]: Not sure what this line is supposed to do?

File: qcfractal/tests/test_server.py [hunk 2]
```diff
@@ -132,3 +133,30 @@ def test_storage_socket(test_server):
...
+
+def test_snowflakehandler_log():
+
+    with FractalSnowflakeHandler() as server:
+        proc = server._qcfractal_proc
+
+        assert "No SSL files passed in" in server.show_log(show=False, nlines=100)
+        assert "0 task" not in server.show_log(show=False, nlines=100)
+
+    assert proc.poll() is not None
```
Comment (Lnaden) [inline qcfractal/tests/test_server.py]: May as well add the empty line here

File: docs/qcfractal/source/setup_server.rst (hunk 1 above)
Comment (dgasmith) [reply docs/qcfractal/source/setup_server.rst]: Accidental leftovers.

**Comment by Lnaden** [comment]
I'll merge this when this passes!

**Pull request merged**